endpoint.go

```diff
@@ -777,7 +766,7 @@ func (ep *endpoint) Delete(force bool) error {
 
 	ep.releaseAddress()
 
-	return nil
+	return n.getEpCnt().DecEndpointCnt()
```

Comment:
Agree but what I am saying is that caller should not treat this as an error and should continue.

For example here https://github.com/docker/docker/blob/master/daemon/container_operations.go#L677 this will cause the entry in `container.NetworkSettings.Networks` not to be removed.

In https://github.com/docker/docker/blob/master/daemon/network.go#L326 it will cause the caller to report an error while instead we know the endpoint was effectively removed and its resources released.

We could return the error as maskable, but I feel the best thing is to return a success and just log an warning in libnetwork.